extension/opampextension/registry_test.go

```diff
@@ -0,0 +1,285 @@
+		case <-coffeeMakerCalledChan: // OK
+		}
+		select {
+		case <-time.After(2 * time.Second):
+			t.Fatalf("Timed out waiting for callback 2 to be called")
+		case <-coffeeMakerCalledChan: // OK
+		}
+	})
+}
+
+func TestCustomCapability_SendMesage(t *testing.T) {
+	t.Run("Sends message", func(t *testing.T) {
+		capabilityString := "io.opentelemetry.teapot"
+		messageType := "brew"
+		mesageBytes := []byte("black")
+
+		client := mockCustomCapabilityClient{
+			sendCustomMessage: func(message *protobufs.CustomMessage) (chan struct{}, error) {
+				require.Equal(t, &protobufs.CustomMessage{
+					Capability: capabilityString,
+					Type:       messageType,
+					Data:       mesageBytes,
+				}, message)
+				return nil, nil
+			},
+		}
+
+		registry := newCustomCapabilityRegistry(zap.NewNop(), client)
+
+		sender, unregister, err := registry.Register(capabilityString, func(_ *protobufs.CustomMessage) {})
+		require.NoError(t, err)
+		require.NotNil(t, sender)
+		require.NotNil(t, unregister)
+
+		channel, err := sender.SendMessage(messageType, mesageBytes)
+		require.NoError(t, err)
+		require.Nil(t, channel, nil)
+	})
+}
+
+func TestCustomCapability_Unregister(t *testing.T) {
+	t.Run("Unregistered capability callback is no longer called", func(t *testing.T) {
+		capabilityString := "io.opentelemetry.teapot"
+		messageType := "steep"
+		mesageBytes := []byte("blackTea")
+		customMessage := &protobufs.CustomMessage{
+			Capability: capabilityString,
+			Type:       messageType,
+			Data:       mesageBytes,
+		}
+
+		client := mockCustomCapabilityClient{}
+
+		registry := newCustomCapabilityRegistry(zap.NewNop(), client)
+
+		unregisteredSender, unregister, err := registry.Register(capabilityString, func(_ *protobufs.CustomMessage) {
+			t.Fatalf("Unregistered capability should not be called")
+		})
+		require.NotNil(t, unregisteredSender)
+		require.NoError(t, err)
+		require.NotNil(t, unregister)
+
+		unregister()
+
+		registry.ProcessMessage(customMessage)
+	})
+
+	t.Run("Unregister is successful even if set capabilities fails", func(t *testing.T) {
+		capabilityString := "io.opentelemetry.teapot"
+		messageType := "steep"
+		mesageBytes := []byte("blackTea")
+		customMessage := &protobufs.CustomMessage{
+			Capability: capabilityString,
+			Type:       messageType,
+			Data:       mesageBytes,
+		}
+
+		client := &mockCustomCapabilityClient{}
+
+		registry := newCustomCapabilityRegistry(zap.NewNop(), client)
+
+		unregisteredSender, unregister, err := registry.Register(capabilityString, func(_ *protobufs.CustomMessage) {
+			t.Fatalf("Unregistered capability should not be called")
+		})
+		require.NotNil(t, unregisteredSender)
+		require.NoError(t, err)
+		require.NotNil(t, unregister)
+
+		client.setCustomCapabilites = func(_ *protobufs.CustomCapabilities) error {
+			return fmt.Errorf("failed to set capabilities")
+		}
+
+		unregister()
+
+		registry.ProcessMessage(customMessage)
+	})
+
+	// FIXME this test is broken
```

Comment:
Is this still accurate? It doesn't look like we are skipping this test.

Reply:
Removed this comment, it was only temporarily broken and I just forgot the comment